streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/kstream/KStream.scala

```diff
@@ -232,10 +232,53 @@ class KStream[K, V](val inner: KStreamJ[K, V]) {
    * @param produced the instance of Produced that gives the serdes and `StreamPartitioner`
    * @return a [[KStream]] that contains the exact same (and potentially repartitioned) records as this [[KStream]]
    * @see `org.apache.kafka.streams.kstream.KStream#through`
+   * @deprecated use `repartition()` instead
    */
+  @deprecated
```

Comment:
We should provide the arguments to the annotation to explain the deprecation. This gives better IDE integration than the ScalaDoc (which we won't need anymore):

```suggestion
  @deprecated("use `repartition()` instead", "2.6.0")
```